hedera-node/hedera-app/src/main/java/com/hedera/node/app/SessionContext.java

```diff
@@ -15,5 +15,22 @@
  */
 package com.hedera.node.app;
 
-// FUTURE: Put the per-thread stuff here like parsers
-public record SessionContext() {}
+import com.google.protobuf.Parser;
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.SignedTransaction;
+import com.hederahashgraph.api.proto.java.Transaction;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+
+/**
+ * This record keeps a list of everything that is used per-thread
+ *
+ * @param queryParser a parser for {@link Query}
+ * @param txParser a parser for {@link Transaction}
+ * @param signedParser a parser for {@link SignedTransaction}
+ * @param txBodyParser a parser for {@link TransactionBody}
+ */
+public record SessionContext(
```

Comment:
null checks?

Reply:
Well, we do not have a definition for it at the moment. And I do not see a difference between a record constructor and a regular constructor when we talk about checks

Reply:
It is maybe a philosophical discussion. Conceptually records are just data-carriers. They aggregate a bunch of values without much ceremony. Therefore I would say, they are not the right tool to implement complex functionality or validate invariants. But anyway, I will change it.